t/udp-generator.pl

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/perl
+
+# Read datagrams from input files and send them to a specified server
+
+use strict;
+use IO::File;
+use IO::Socket;
+
+if (scalar @ARGV < 3) {
+	print STDERR "Usage: $0 [server] [port] [input...]\n";
+	exit 1;
+}
+
+my $server = shift @ARGV;
+my $port = shift @ARGV;
+
+my $sock = IO::Socket::INET->new(
+	Type     => SOCK_DGRAM,
+	Proto    => 'udp',
+	PeerAddr => $server,
+	PeerPort => $port,
+) or die "Failed to create a UDP socket: $!\n";
+
+foreach my $f (@ARGV) {
+	my $fh = IO::File->new($f, "r") or die "Failed to open file: $!\n";
```

Comment:
```suggestion
	my $fh = IO::File->new($f, "r") or die "Failed to open file: $f:$!";
```
Maybe print the filename as well?

Also, generally speaking it's good to omit newline at the end of `die` or `warn`, because then perl would emit the location within the source code at where the problem occurred. That applies to other parts using `die` as well.

Reply:
Fixed in 9a355c6 and 65f477e.